automated-tests/src/main/java/com/hp/ov/sdk/settings/ScopeResource.java

```diff
@@ -0,0 +1,124 @@
+import com.hp.ov.sdk.oneview.CreateResource;
+import com.hp.ov.sdk.oneview.PatchResource;
+import com.hp.ov.sdk.oneview.RemoveResource;
+import com.hp.ov.sdk.oneview.UpdateResource;
+import com.hp.ov.sdk.rest.client.networking.EthernetNetworkClient;
+import com.hp.ov.sdk.rest.client.networking.FcNetworkClient;
+import com.hp.ov.sdk.rest.client.settings.ScopeClient;
+
+public class ScopeResource extends BasicResource implements CreateResource, RemoveResource, UpdateResource, PatchResource {
+
+    private static ScopeResource instance;
+    private ScopeClient client;
+    private final FcNetworkClient fcNetworkClient;
+    private final EthernetNetworkClient ethernetNetworkClient;
+
+    public ScopeResource(){
+        client = oneViewClient.scope();
+        this.fcNetworkClient = oneViewClient.fcNetwork();
+        this.ethernetNetworkClient = oneViewClient.ethernetNetwork();
+    }
+
+    public static ScopeResource getInstance() {
+        if (instance == null) {
+            instance = new ScopeResource();
+        }
+        return instance;
+    }
+
+    @Override
+    public Map<String, String> getResourceProperties(String id){
+        return getResourceProperties(client.getById(id));
+    }
+
+    @Override
+    public String findByName(String name) {
+        Scope scope = (Scope) getResource(client.getByName(name));
+        return scope == null ? "" : scope.getResourceId();
+    }
+
+    @Override
+    public String findById(String id) {
+        try {
+            return client.getById(id).getName();
+        } catch (final SDKResourceNotFoundException ex) {
+            return "";
+        }
+    }
+
+    @Override
+    public int count() {
+        return getCount(client.getAll());
+    }
+
+    @Override
+    public void create() {
+        client.create(builder());
+    }
+
+    @Override
+    public String patch(String id){
+        return taskToString(client.patch(id, builderPatch(client.getById(id))));
+    }
+
+    @Override
+    public String update(String id){
+        return objToString(client.update(id, builderUpdate(client.getById(id))));
+    }
+
+    @Override
+    public String remove(String id) {
+        return taskToString(client.delete(id));
+    }
+
+    public Scope builder() {
+        Scope scope = new Scope();
+        scope.setType(Scope.TYPE);
+        scope.setName(resourceProperties.get("name"));
+        scope.setDescription(resourceProperties.get("description"));
+        return scope;
+    }
+
+    private Scope builderUpdate(Scope scope) {
+        scope.setName(resourceProperties.get("name"));
+        scope.setDescription(resourceProperties.get("description"));
+        return scope;
+    }
+
+    private ResourceAssignment builderPatch(Scope scope) {
+        FcNetwork fcNetwork = this.fcNetworkClient.getByName(resourceProperties.get("fcNetwork")).get(0);
+        Network ethernetNetwork = this.ethernetNetworkClient.getByName(resourceProperties.get("ethernetNetwork")).get(0);
```

Comment:
Just keep one resource type here, and make it default. So dont name it FcNetwork, or ethernetNetwork, because it can be any resource type we want. Make it more general.